internal/groups/postgres/groups.go

```diff
@@ -146,17 +144,21 @@ func (repo groupRepository) RetrieveByID(ctx context.Context, id string) (mggrou
 }
 
 func (repo groupRepository) RetrieveAll(ctx context.Context, gm mggroups.Page) (mggroups.Page, error) {
-	var q string
-	query := buildQuery(gm)
+	var query, q, count string
 
 	if gm.ID != "" {
-		q = buildHierachy(gm)
+		q, count = buildHierachy(gm)
+		query = buildQuery(gm, "AND")
+
+		q = fmt.Sprintf(" %s %s ORDER BY g.created_at LIMIT :limit OFFSET :offset;", q, query)
 	}
 	if gm.ID == "" {
 		q = `SELECT DISTINCT g.id, g.domain_id, COALESCE(g.parent_id, '') AS parent_id, g.name, g.description,
-		g.metadata, g.created_at, g.updated_at, g.updated_by, g.status FROM groups g`
+		g.metadata, g.created_at, g.updated_at, g.updated_by, g.status, nlevel(path) as level, g.path FROM groups g`
+		query = buildQuery(gm, "WHERE")
```

Comment:
Add the `WHERE` and `AND` clauses in buildquery

Reply:
They are added in buildquery @rodneyosodo, I added `WHERE` as a parameter since at times we need to build the query with a `WHERE` clause and at times we need to append more queries using `AND`.